autopush/endpoint.py

```diff
@@ -25,6 +28,41 @@
 
 All message bodies must be UTF-8 encoded.
 
+Response Format
+---------------
+
+All responses will be of the content-type `application/json`, the structure of
+which depends on whether the requests was successful or not. In the case of
+success, the status code will be in the 2XX family.
+
+Error responses will be in the following format:
+
+.. code-block:: json
+
+    {
+        "code": 404, // matches the HTTP status code
+        "errno": 103, // stable application-level error number
+        "error": "Not Found", // string representation of the status
+        "message": "No message found" // Optional additional error information
+    }
+
+The current application-level error numbers are:
+
+* status code 400, errno 101: missing necessary crypto keys
+* status code 404, errno 102: invalid URL endpoint
+* status code 404, errno 103: expired URL endpoint
+* status code 413, errno 104: data payload is too large
+* status code 404, errno 105: endpoint became unavailable during request
+* status code 404, errno 106: invalid subscription
+* status code 404, errno 107: message not found
+* status code 400, errno 108: router type is invalid
+* status code 401, errno 109: invalid authentication
```

Comment:
seems redundant to have a discrete errno for an existing well defined status response, but whatever.